From ef4e88e5c8aee247a6dc54c97773e18a407182f1 Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Tue, 18 Nov 2014 08:06:41 +0100
Subject: [PATCH 1/3] Load libEGL and libGLESv2 at the right version

In some systems the plain .so files without a version number are not available and
therefor we have to use the versioned ones. For both we can safely assume to use version 1
for EGL and 2 for GLESv2 by default.

Signed-off-by: Simon Busch <morphis@gravedo.de>
---
 src/core/surface_factory_qt.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/core/surface_factory_qt.cpp b/src/core/surface_factory_qt.cpp
index 40c65d9..db10be0 100644
--- a/src/core/surface_factory_qt.cpp
+++ b/src/core/surface_factory_qt.cpp
@@ -71,13 +71,13 @@ base::NativeLibrary LoadLibrary(const base::FilePath& filename) {
 bool SurfaceFactoryQt::LoadEGLGLES2Bindings(AddGLLibraryCallback add_gl_library, SetGLGetProcAddressProcCallback set_gl_get_proc_address)
 {
     base::FilePath libEGLPath = QtWebEngineCore::toFilePath(QT_LIBDIR_EGL);
-    libEGLPath = libEGLPath.Append("libEGL.so");
+    libEGLPath = libEGLPath.Append("libEGL.so.1");
     base::NativeLibrary eglLibrary = LoadLibrary(libEGLPath);
     if (!eglLibrary)
         return false;
 
     base::FilePath libGLES2Path = QtWebEngineCore::toFilePath(QT_LIBDIR_GLES2);
-    libGLES2Path = libGLES2Path.Append("libGLESv2.so");
+    libGLES2Path = libGLES2Path.Append("libGLESv2.so.2");
     base::NativeLibrary gles2Library = LoadLibrary(libGLES2Path);
     if (!gles2Library)
         return false;
-- 
2.1.4

